diff --git a/components/signin/internal/identity_manager/gaia_cookie_manager_service.cc b/components/signin/internal/identity_manager/gaia_cookie_manager_service.cc
index e1157df0469281b3563fe88cf2589f25073e8612..1798f24f27775e167ffef3d12bac07dc25c1e451 100644
--- a/components/signin/internal/identity_manager/gaia_cookie_manager_service.cc
+++ b/components/signin/internal/identity_manager/gaia_cookie_manager_service.cc
@@ -553,6 +553,7 @@ void GaiaCookieManagerService::AddAccountToCookieWithToken(
 bool GaiaCookieManagerService::ListAccounts(
     std::vector<gaia::ListedAccount>* accounts,
     std::vector<gaia::ListedAccount>* signed_out_accounts) {
+  return true;
   if (accounts)
     accounts->assign(listed_accounts_.begin(), listed_accounts_.end());
 
