diff --git a/components/password_manager/core/browser/login_database_posix.cc b/components/password_manager/core/browser/login_database_posix.cc
index 57708cce77197b17059b58e602c558be55c29679..ed3e3ae7b149c38f2868f0707ce120ee8ff552e8 100644
--- a/components/password_manager/core/browser/login_database_posix.cc
+++ b/components/password_manager/core/browser/login_database_posix.cc
@@ -81,7 +81,7 @@ LoginDatabase::EncryptionResult LoginDatabase::DecryptedString(
                                      ? PasswordDecryptionResult::kSucceeded
                                      : PasswordDecryptionResult::kFailed);
   return decryption_success ? ENCRYPTION_RESULT_SUCCESS
-                            : ENCRYPTION_RESULT_SERVICE_FAILURE;
+                            : ENCRYPTION_RESULT_ITEM_FAILURE;
 }
 
 }  // namespace password_manager
